tests/suites/test_suite_psa_crypto.function

```diff
@@ -9301,6 +9301,103 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE */
+void derive_verify_bytes(int alg_arg,
+                         data_t *key_data, data_t *input1, data_t *input2,
+                         int requested_capacity_arg,
+                         data_t *expected_output, int expected_output_status)
+{
+    mbedtls_svc_key_id_t key = MBEDTLS_SVC_KEY_ID_INIT;
+    psa_algorithm_t alg = alg_arg;
+    size_t requested_capacity = requested_capacity_arg;
+    psa_key_derivation_operation_t operation = PSA_KEY_DERIVATION_OPERATION_INIT;
+
+    psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
+
+    PSA_ASSERT(psa_crypto_init());
+
+    psa_set_key_usage_flags(&attributes, PSA_KEY_USAGE_DERIVE);
```

Comment:
I think it would be a good idea to parametrise the usage flags, and pass them in with the test data, as [in the future ](https://github.com/Mbed-TLS/mbedtls/issues/8446) we likely want to test different flags. @gilles-peskine-arm do you agree?